planner/core/logical_plan_builder.go

```diff
@@ -428,6 +429,7 @@ func (ds *DataSource) setPreferredStoreType(hintInfo *tableHintInfo) {
 	}
 	if hintInfo.ifPreferTiFlash(alias) {
 		ds.preferStoreType |= preferTiFlash
+		ds.possibleAccessPaths = append(ds.possibleAccessPaths, &accessPath{isTablePath: true, storeType: kv.TiFlash})
```

Comment:
Can we do it when build possibleAccessPaths? From the function name, it should not change access paths.

Reply:
If the `TiFlash` replica is not available. We can't generate a flash access path. But when the hint is set. we should enforce a flash access path.

Reply:
@winoros That's not a temp behavior. For now, CBO can judge whether to choose the TiFlash access path by `tblInfo.TiFlashReplica.Available`. I hope when users set the hint, the planner enforce a TiFlash path without check the TiFlash replica status.

Reply:
Hint should not enforce something that tidb cannot do, it is only a suggestion for optimizer. Therefore, if the tiflash access path is not available, optimizer should not use it even if there are hints.

Reply:
But we don't know whether the TiFlash access path is available. The table info of TiFlash replica is inaccurate information. For example, the user doesn't create the TiFlash replica by DDL or the respond of replica status is missed by the network. If the planner can be enforced by hint, then the TiFlash replica is always unavailable.

Reply:
`the respond of replica status is missed by the network` This should not be a normal state?

Reply:
Yes, it's a normal performance. But I still want to leave a function to force accessing the TiFlash. It's easy for both PD and TiFlash to check the bug when the replica is unavailable.

Reply:
There would be 2 TiFlash paths then? since `getPossibleAccessPaths` may has appended one already.

Reply:
Yes, but it doesn't affect the correctness. Just make the planner a bit slower.

Reply:
I prefer to check if TiFlash path exists before adding a new one. Also, please add comment to explain that this change is for testing purpose, and add a TODO to remind reverting it finally.

Reply:
> I prefer to check if TiFlash path exists before adding a new one. Also, please add comment to explain that this change is for testing purpose, and add a TODO to remind reverting it finally.

@eurekaka Done.